### PR TITLE
Implements setting to disable safe spawning

### LINKED ISSUE #500: Add an option to keep the player in the same the exact position when relogging
# ❇ Feature Request

### 📸 Screenshots / Videos

https://youtu.be/xK2NZq21gz4

### ✔ Expected Behavior

After you relog, you will be in the exact same position where you left the server, even if it's a deadly place.

### ❌ Actual Behavior

You get teleported to a safe position. Players can use it to escape the void or dangerous places like inside a lava lake. And can also get teleported into odd places for no reason (like what happened in the video)

### 💬 Anything else we should know?

Currently at: https://debugpaste.powernukkit.org/pastes/soliledacu

### PR DESCRIPTION
**Pull request by RORIdev**
Fixes #500

### CODE REVIEW

File: src/main/java/cn/nukkit/level/Level.java [hunk 1]
```diff
@@ -3264,6 +3264,9 @@ public Position getSafeSpawn(Vector3 spawn) {
         }
 
         if (spawn != null) {
+            if(!server.isSafeSpawn()) {
+                return new Position(spawn.x,spawn.y,spawn.z);
+            }
```
Comment (AntonPPDS) [inline src/main/java/cn/nukkit/level/Level.java:3269]: Plugins can use this method to find a safe position to teleport the player. Rather, this check should be done where the getSafeSpawn() method is used, not here

**Review by AntonPPDS — CHANGES_REQUESTED** [review]

File: src/main/java/cn/nukkit/level/Level.java (hunk 1 above)
Comment (RORIdev) [reply src/main/java/cn/nukkit/level/Level.java:3269]: Ah sure, i just looked at the source of the issue, and my debbugger was completely broken. will fix.

Comment (RORIdev) [reply src/main/java/cn/nukkit/level/Level.java]: Moved to calling side.

**Review by joserobjr — APPROVED** [review]
Thank you! 👾

**Pull request merged**